### DISCUSSION TITLE
Set UI5 Input value show [UI5-FWK] numeric value for property [value] of component [ui5-input] is missing "{ type: Number }" in its property decorator.

### DISCUSSION THREAD

**Discussion by pupss89345** [UI5/webcomponents#10072] [Q&A] [answered]
Hi I am use UI5-Input And when I use button click set a numeric value to input is show  numeric value for property [value] of component [ui5-input] is missing "{ type: Number }" in its property decorator ,but My input has already set type as Number .
I don't know why show that ??

My HTML

```
<ui5-button id="bu" click="OnE_Datechenged">Press</ui5-button>
<ui5-input id="dd" value="0" type="Number"></ui5-input>
```

JS

```
const button = document.getElementById("bu");

button.addEventListener("click", (event) => {

   document.getElementById("dd").value=5*3;
});
```

error
![image](https://github.com/user-attachments/assets/844d5f41-1c21-44da-a635-57a01fdf1120)

![動畫2](https://github.com/user-attachments/assets/61dd4ad0-f955-4f4e-a441-cdb4e36f3023)

**Comment by pupss89345** [comment, marked as answer]
set value as string type is OK ....

**Comment by pskelin** [comment]
Yes your finding is correct. This warning is about the type of the property, not the type of the input.

In the Input.ts, the `value` property is of type string (both typescript and for the framework)
```ts
@property()
value = "";
```

This is a dev-time check, if at runtime a number is assigned to a property, it could be that the developer of the component wanted to make a number property and then it should be:
```ts
@property({ type: Number })
step = 0;
```

This is important for attribute conversion that the correct type is set. You encounter this error as an app developer as you are using pure javascript. If you were to use TSX via the react wrappers, then typescript would give you the exact same warning
<img width="984" alt="Screenshot 2024-11-01 at 9 01 00" src="https://github.com/user-attachments/assets/6e55c75c-9e27-4004-9215-2e77efc92aab">

And then the fix is exactly the same - you have to convert the number to a string before assigning it.
